viewer/src/hooks/useTracesData.js

```diff
@@ -0,0 +1,26 @@
+import React, { useState, useEffect, useContext } from "react";
+import FetchTracesQueryContext from "../contexts/FetchTracesQueryContext";
+import { useQuery } from '@tanstack/react-query'
+import { fetchTracesData } from "../queries/tracesData";
+
+export const useTracesData = (projectId, traceNames) => {
+    const fetchTraceQuery = useContext(FetchTracesQueryContext)
```

Comment:
There will be a difference FetchTraceQueryContext provider in the hosted version.